sync/src/types.rs

```diff
@@ -280,11 +280,90 @@ impl From<(BlockNumber, Byte32)> for BlockNumberAndHash {
     }
 }
 
+enum SchedulerOutput {
+    DoubleInCrease,
```

Comment:
inconsistent naming style, `InCrease` vs `Decrease` vs `DoubleDeCrease`
```suggestion
    DoubleIncrease,
    Increase,
    Decrease,
    DoubleDecrease
```

and I don't like the enum name here, it's a quantile, not related to scheduler, how about renaming the `DynamicTimeLimit` to `TimeAnalyzer`, and return a `TimeQuantile` enum?
```
MinToLower
LowerToMedian
MedianToUpper
UpperToMax
```